# Naive Baselines — Weather Forecasting for Spanish SYNOP Stations

This notebook establishes naive baselines for temperature and precipitation forecasting over four prototype SYNOP stations. These baselines define the minimum performance threshold that any trained model must exceed to demonstrate genuine predictive value.

**Forecast targets:** daily maximum temperature (tmax), minimum temperature (tmin), and precipitation (prec)  
**Forecast horizons:** t+1, t+3, t+7 days  
**Prototype stations:** Córdoba Aeropuerto, Burgos, Puerto de Navacerrada, Palma Puerto  
**Validation scheme:** Expanding Window CV — 20 annual windows, minimum 15 years of training data, identical to the ML baseline scheme

## 1. Baseline Methods

Two naive baselines are evaluated:

**Persistence:** the forecast for day t+h is the observed value at day t — the last known observation before the forecast horizon. This is the simplest possible baseline and captures the short-term autocorrelation of meteorological variables.

**Climatology (seasonal mean):** the forecast for any given day is the historical mean for that calendar day, computed exclusively from the training data of each fold. This captures the annual cycle without using any recent observations.

Neither baseline involves any model training. Their purpose is to establish a lower performance bound: a model that does not beat both baselines provides no useful information beyond what is already available from historical patterns.

## 2. Results — tmax (°C)

In [1]:
import pandas as pd

tmax_data = [
    ('Córdoba',     't+1', 1.994, 0.909, 3.009, 0.813),
    ('Córdoba',     't+3', 3.188, 0.779, 3.009, 0.813),
    ('Córdoba',     't+7', 3.824, 0.684, 3.009, 0.813),
    ('Burgos',      't+1', 2.712, 0.829, 3.641, 0.723),
    ('Burgos',      't+3', 4.160, 0.619, 3.641, 0.723),
    ('Burgos',      't+7', 4.747, 0.518, 3.641, 0.723),
    ('Navacerrada', 't+1', 2.540, 0.858, 4.028, 0.682),
    ('Navacerrada', 't+3', 4.214, 0.627, 4.027, 0.682),
    ('Navacerrada', 't+7', 5.077, 0.473, 4.029, 0.683),
    ('Palma',       't+1', 1.599, 0.869, 2.009, 0.800),
    ('Palma',       't+3', 2.139, 0.773, 2.009, 0.800),
    ('Palma',       't+7', 2.494, 0.693, 2.009, 0.800),
]

cols = ['Station', 'Horizon',
        'Persistence MAE', 'Persistence R²',
        'Climatology MAE', 'Climatology R²']

df_tmax = pd.DataFrame(tmax_data, columns=cols)
display(df_tmax)

,Station,Horizon,Persistence MAE,Persistence R²,Climatology MAE,Climatology R²
0,Córdoba,t+1,1.994,0.909,3.009,0.813
1,Córdoba,t+3,3.188,0.779,3.009,0.813
2,Córdoba,t+7,3.824,0.684,3.009,0.813
3,Burgos,t+1,2.712,0.829,3.641,0.723
4,Burgos,t+3,4.160,0.619,3.641,0.723
5,Burgos,t+7,4.747,0.518,3.641,0.723
6,Navacerrada,t+1,2.540,0.858,4.028,0.682
7,Navacerrada,t+3,4.214,0.627,4.027,0.682
8,Navacerrada,t+7,5.077,0.473,4.029,0.683
9,Palma,t+1,1.599,0.869,2.009,0.800


## 3. Results — tmin (°C)

In [2]:
tmin_data = [
    ('Córdoba',     't+1', 1.761, 0.870, 2.408, 0.778),
    ('Córdoba',     't+3', 2.593, 0.729, 2.408, 0.778),
    ('Córdoba',     't+7', 2.999, 0.643, 2.409, 0.778),
    ('Burgos',      't+1', 2.108, 0.758, 2.695, 0.627),
    ('Burgos',      't+3', 3.148, 0.471, 2.695, 0.628),
    ('Burgos',      't+7', 3.561, 0.346, 2.696, 0.629),
    ('Navacerrada', 't+1', 1.884, 0.854, 3.200, 0.646),
    ('Navacerrada', 't+3', 3.431, 0.564, 3.200, 0.647),
    ('Navacerrada', 't+7', 4.090, 0.404, 3.203, 0.647),
    ('Palma',       't+1', 1.259, 0.914, 1.780, 0.844),
    ('Palma',       't+3', 1.899, 0.815, 1.779, 0.844),
    ('Palma',       't+7', 2.202, 0.757, 1.778, 0.844),
]

df_tmin = pd.DataFrame(tmin_data, columns=cols)
display(df_tmin)

,Station,Horizon,Persistence MAE,Persistence R²,Climatology MAE,Climatology R²
0,Córdoba,t+1,1.761,0.870,2.408,0.778
1,Córdoba,t+3,2.593,0.729,2.408,0.778
2,Córdoba,t+7,2.999,0.643,2.409,0.778
3,Burgos,t+1,2.108,0.758,2.695,0.627
4,Burgos,t+3,3.148,0.471,2.695,0.628
5,Burgos,t+7,3.561,0.346,2.696,0.629
6,Navacerrada,t+1,1.884,0.854,3.200,0.646
7,Navacerrada,t+3,3.431,0.564,3.200,0.647
8,Navacerrada,t+7,4.090,0.404,3.203,0.647
9,Palma,t+1,1.259,0.914,1.780,0.844


## 4. Results — Precipitation (mm)

In [3]:
prec_data = [
    ('Córdoba',     't+1', 2.038, -0.335, 2.570, -0.046),
    ('Córdoba',     't+3', 2.509, -0.761, 2.571, -0.047),
    ('Córdoba',     't+7', 2.673, -0.887, 2.570, -0.046),
    ('Burgos',      't+1', 1.961, -0.405, 2.261, -0.040),
    ('Burgos',      't+3', 2.473, -0.842, 2.260, -0.039),
    ('Burgos',      't+7', 2.601, -0.937, 2.261, -0.037),
    ('Navacerrada', 't+1', 4.382, -0.271, 5.023, -0.016),
    ('Navacerrada', 't+3', 5.560, -0.766, 5.022, -0.015),
    ('Navacerrada', 't+7', 5.978, -0.903, 5.019, -0.014),
    ('Palma',       't+1', 1.995, -0.531, 2.215, -0.033),
    ('Palma',       't+3', 2.386, -0.902, 2.216, -0.033),
    ('Palma',       't+7', 2.439, -0.945, 2.211, -0.033),
]

cols_prec = ['Station', 'Horizon',
             'Persistence MAE', 'Persistence R²',
             'Climatology MAE', 'Climatology R²']

df_prec = pd.DataFrame(prec_data, columns=cols_prec)
display(df_prec)

,Station,Horizon,Persistence MAE,Persistence R²,Climatology MAE,Climatology R²
0,Córdoba,t+1,2.038,-0.335,2.570,-0.046
1,Córdoba,t+3,2.509,-0.761,2.571,-0.047
2,Córdoba,t+7,2.673,-0.887,2.570,-0.046
3,Burgos,t+1,1.961,-0.405,2.261,-0.040
4,Burgos,t+3,2.473,-0.842,2.260,-0.039
5,Burgos,t+7,2.601,-0.937,2.261,-0.037
6,Navacerrada,t+1,4.382,-0.271,5.023,-0.016
7,Navacerrada,t+3,5.560,-0.766,5.022,-0.015
8,Navacerrada,t+7,5.978,-0.903,5.019,-0.014
9,Palma,t+1,1.995,-0.531,2.215,-0.033


## 5. Key Findings

### 5.1 Temperature: a clear crossover between baselines

A consistent pattern emerges across all four stations and both temperature variables:

- **At t+1**, persistence outperforms climatology in every station. Temperature has strong short-term autocorrelation: tomorrow will closely resemble today, and persistence captures that signal directly.
- **At t+3 and t+7**, climatology outperforms persistence in every station. As the forecast horizon increases, the memory of current conditions fades and the seasonal cycle dominates. Climatology encodes that seasonal signal by design.

This crossover defines the **predictability horizon of naive methods**: somewhere between t+1 and t+3, short-term memory becomes less useful than long-term seasonal patterns.

The implication for model evaluation is direct: any trained model must beat persistence at t+1 and climatology at t+3 and t+7. The harder bar varies by horizon.

### 5.2 LightGBM clears both baselines at all horizons

The ML baseline (LightGBM) outperforms both naive baselines for temperature across all stations and all horizons. The margin over climatology narrows at t+7 — particularly for Burgos (LightGBM 3.58°C vs climatology 3.64°C) — but LightGBM consistently remains below both naive thresholds. This validates that the ML models extract genuine predictive signal beyond naive pattern-matching.

### 5.3 Precipitation: naive baselines fail entirely

Both baselines produce negative R² for precipitation across all stations and all horizons. A negative R² means the baseline performs worse than simply predicting the sample mean — a result that reflects the physical nature of daily precipitation rather than a methodological failure.

Precipitation is governed by mesoscale and synoptic atmospheric dynamics (frontal systems, convective triggering, orographic lifting) that are not captured by local station observations alone. Neither the memory of yesterday's rain nor the seasonal average provides a reliable signal for tomorrow's precipitation at daily resolution.

This finding motivates two directions in subsequent phases:
1. **Deep learning:** LSTM and GRU architectures may capture sequential dependencies in station data that naive methods cannot.
2. **Problem reformulation:** if DL does not substantially improve precipitation forecasting, the task will be reformulated as binary classification (rain / no rain) with a physically motivated threshold.

## 6. Reference Thresholds for Model Evaluation

The table below summarises the hardest naive baseline per station, variable and horizon — the minimum bar that any trained model must beat.

In [4]:
thresholds = [
    # Temperature — persistence wins at t+1, climatology at t+3 and t+7
    ('Córdoba',     'tmax', 't+1', 'Persistence', 1.994),
    ('Córdoba',     'tmax', 't+3', 'Climatology', 3.009),
    ('Córdoba',     'tmax', 't+7', 'Climatology', 3.009),
    ('Córdoba',     'tmin', 't+1', 'Persistence', 1.761),
    ('Córdoba',     'tmin', 't+3', 'Climatology', 2.408),
    ('Córdoba',     'tmin', 't+7', 'Climatology', 2.409),
    ('Burgos',      'tmax', 't+1', 'Persistence', 2.712),
    ('Burgos',      'tmax', 't+3', 'Climatology', 3.641),
    ('Burgos',      'tmax', 't+7', 'Climatology', 3.641),
    ('Burgos',      'tmin', 't+1', 'Persistence', 2.108),
    ('Burgos',      'tmin', 't+3', 'Climatology', 2.695),
    ('Burgos',      'tmin', 't+7', 'Climatology', 2.696),
    ('Navacerrada', 'tmax', 't+1', 'Persistence', 2.540),
    ('Navacerrada', 'tmax', 't+3', 'Climatology', 4.027),
    ('Navacerrada', 'tmax', 't+7', 'Climatology', 4.029),
    ('Navacerrada', 'tmin', 't+1', 'Persistence', 1.884),
    ('Navacerrada', 'tmin', 't+3', 'Climatology', 3.200),
    ('Navacerrada', 'tmin', 't+7', 'Climatology', 3.203),
    ('Palma',       'tmax', 't+1', 'Persistence', 1.599),
    ('Palma',       'tmax', 't+3', 'Climatology', 2.009),
    ('Palma',       'tmax', 't+7', 'Climatology', 2.009),
    ('Palma',       'tmin', 't+1', 'Persistence', 1.259),
    ('Palma',       'tmin', 't+3', 'Climatology', 1.779),
    ('Palma',       'tmin', 't+7', 'Climatology', 1.778),
]

df_thresh = pd.DataFrame(thresholds,
    columns=['Station', 'Variable', 'Horizon', 'Hardest Baseline', 'MAE Threshold (°C)'])
display(df_thresh)

,Station,Variable,Horizon,Hardest Baseline,MAE Threshold (°C)
0,Córdoba,tmax,t+1,Persistence,1.994
1,Córdoba,tmax,t+3,Climatology,3.009
2,Córdoba,tmax,t+7,Climatology,3.009
3,Córdoba,tmin,t+1,Persistence,1.761
4,Córdoba,tmin,t+3,Climatology,2.408
5,Córdoba,tmin,t+7,Climatology,2.409
6,Burgos,tmax,t+1,Persistence,2.712
7,Burgos,tmax,t+3,Climatology,3.641
8,Burgos,tmax,t+7,Climatology,3.641
9,Burgos,tmin,t+1,Persistence,2.108


**Note on DL comparison scope**

The naive baselines above cover t+1, t+3 and t+7 — the horizons evaluated in the ML baseline phase. The LSTM model predicts all horizons from t+1 to t+7 simultaneously. Direct comparison against naive and ML baselines is available at t+1, t+3 and t+7 only. Results at t+2, t+4, t+5 and t+6 are reported as DL-only outputs without a reference baseline.